apiserver/facades/agent/provisioner/provisioninginfo.go

```diff
@@ -118,10 +124,34 @@ func (api *ProvisionerAPI) getProvisioningInfoBase(m *state.Machine,
 		EndpointBindings: endpointBindings,
 	}
 
-	var err error
-	if result.Constraints, err = m.Constraints(); err != nil {
+	machineCons, err := m.Constraints()
+	if err != nil {
 		return result, errors.Trace(err)
 	}
+	cons := params.Value{
+		Arch:             machineCons.Arch,
+		Container:        machineCons.Container,
+		CpuCores:         machineCons.CpuCores,
+		CpuPower:         machineCons.CpuPower,
+		Mem:              machineCons.Mem,
+		RootDisk:         machineCons.RootDisk,
+		RootDiskSource:   machineCons.RootDiskSource,
+		Tags:             machineCons.Tags,
+		InstanceRole:     machineCons.InstanceRole,
+		InstanceType:     machineCons.InstanceType,
+		Spaces:           machineCons.Spaces,
+		VirtType:         machineCons.VirtType,
+		Zones:            machineCons.Zones,
+		AllocatePublicIP: machineCons.AllocatePublicIP,
+		ImageID:          machineCons.ImageID,
+	}
+	result.Constraints = cons
+
+	// Hardcode charm mem constraints to 64Mi, limit to 256Mi.
+	result.CharmConstraints = params.CharmValue{
```

Comment:
Machines don't use charm constraints.
Also we don't want to change the existing constraints struct for this PR.

Reply:
reverted this per discussion offline